plugins/imfile/imfile.c

```diff
@@ -746,13 +758,44 @@ poll_active_files(fs_edge_t *const edge)
 	}
 }
 
+static rsRetVal ATTR_NONNULL()
+process_symlink(fs_edge_t *const chld, const char *symlink)
+{
+	DEFiRet;
+	char *target = NULL;
+	CHKmalloc(target = realpath(symlink, target));
+	struct stat fileInfo;
+	if(stat(target, &fileInfo) != 0) {
+		LogError(errno, RS_RET_ERR,	"imfile: process_symlink cannot stat file '%s' - ignored", target);
+		FINALIZE;
+	}
+	const int is_file = (S_ISREG(fileInfo.st_mode));
+	DBGPRINTF("process_symlink:  found '%s', File: %d (config file: %d), symlink: %d\n",
+		target, is_file, chld->is_file, 0);
+	if(!is_file && S_ISREG(fileInfo.st_mode)) {
+		LogMsg(0, RS_RET_ERR, LOG_WARNING,
+			"imfile: '%s' is neither a regular file, symlink, nor a directory - ignored", target);
+		FINALIZE;
+	}
+	if(chld->is_file != is_file) {
+		LogMsg(0, RS_RET_ERR, LOG_WARNING,
+			"imfile: '%s' is %s but %s expected - ignored",
+			target, (is_file) ? "FILE" : "DIRECTORY", (chld->is_file) ? "FILE" : "DIRECTORY");
+		FINALIZE;
+	}
+	act_obj_add(chld, target, is_file, fileInfo.st_ino, 0, symlink);
+
+finalize_it:
+	free(target);
+	RETiRet;
+}
 
-static void ATTR_NONNULL() poll_tree(fs_edge_t *const chld);
```

Comment:
I think I remember this line was required by some analyzer - but I may be wrong. CI will hopefully show,

Reply:
I think that last run almost past (minus the extra file left but actual tests were green)